### PR TITLE
spinal.lib.bus.fabric reusable base classes

### PR DESCRIPTION
**Pull request by Dolu1990**
Another attempts of #1156
WIP

### CODE REVIEW

**Comment by kleinai** [comment]
There's a whole thread in TileLink node that's not implemented in `bus.fabric`. Is that assumed to be implementation specific?

**Comment by Dolu1990** [comment]
> There's a whole thread in TileLink node that's not implemented in `bus.fabric`. Is that assumed to be implementation specific?

Yes, the idea is to keep it for the final class.

Also, note that for AXI, there will be no s2m, only m2s.

File: lib/src/main/scala/spinal/lib/bus/fabric/NegociateSP.scala [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+import spinal.core._
+import spinal.core.fiber.{Handle, soon}
+
+import scala.collection.mutable.ArrayBuffer
+
+/**
+  * Negotiation handles for master to slave requests
+  */
+abstract class NegociateSP[S, P] extends Area{
```
Comment (kleinai) [inline lib/src/main/scala/spinal/lib/bus/fabric/NegociateSP.scala:11]: Quick fix: this should be `NegotiateSP`

Comment (Dolu1990) [reply lib/src/main/scala/spinal/lib/bus/fabric/NegociateSP.scala:11]: Sure :D

**Comment by Dolu1990** [comment]
I guess the dev can continue in the bus-fabric branch

@kleinai Were you thinking about implementing a AXI version ?

**Comment by kleinai** [comment]
> I guess the dev can continue in the bus-fabric branch
> 
> @kleinai Were you thinking about implementing a AXI version ?

That was the plan to test out the new changes.

**Comment by Dolu1990** [comment]
> > I guess the dev can continue in the bus-fabric branch
> > @kleinai Were you thinking about implementing a AXI version ?
> 
> That was the plan to test out the new changes.

Cool ^^
Maybe we should do a call then :)

**Pull request merged**